envbuilder.go

```diff
@@ -1010,7 +1010,7 @@ func RunCacheProbe(ctx context.Context, opts options.Options) (v1.Image, error)
 					opts.Logger(log.LevelInfo, "No Dockerfile or image specified; falling back to the default image...")
 					fallbackDockerfile = defaultParams.DockerfilePath
 				}
-				buildParams, err = devContainer.Compile(opts.Filesystem, devcontainerDir, constants.MagicDir, fallbackDockerfile, opts.WorkspaceFolder, false, os.LookupEnv)
+				buildParams, err = devContainer.Compile(opts.Filesystem, devcontainerDir, buildTimeWorkspaceFolder, fallbackDockerfile, opts.WorkspaceFolder, false, os.LookupEnv)
```

Comment:
Will this potentially change any of the Docker instructions/"directives"? That could lead to a cache miss if that's the case. Otherwise LGTM.

Reply:
I see a comment in `devcontainer.go` that mentions `featureDir` must be reproducible based on the feature name, but I think that's relative to the build context.

In any case, I think it makes sense to beef up the testing around `RunCacheProbe` outside of a container as a non-root user.

Reply:
I looked into adding some more non-rootful cache probe tests and I then remembered that you actually need the envbuilder binary for this. The infrastructure for all that is already there in the terraform provider tests, so that's probably a better place to do it.